server/src/main/java/org/activityinfo/server/endpoint/akvo/flow/AkvoFlowResource.java

```diff
@@ -0,0 +1,147 @@
+package org.activityinfo.server.endpoint.akvo.flow;
+
+import com.google.inject.Inject;
+import liquibase.util.csv.CSVReader;
+import org.activityinfo.legacy.shared.command.CreateEntity;
+import org.activityinfo.legacy.shared.command.GetSchema;
+import org.activityinfo.legacy.shared.command.UpdateFormClass;
+import org.activityinfo.legacy.shared.command.result.CreateResult;
+import org.activityinfo.legacy.shared.model.ActivityFormDTO;
+import org.activityinfo.legacy.shared.model.LocationTypeDTO;
+import org.activityinfo.legacy.shared.model.UserDatabaseDTO;
+import org.activityinfo.model.form.FormClass;
+import org.activityinfo.model.form.FormField;
+import org.activityinfo.model.legacy.CuidAdapter;
+import org.activityinfo.model.resource.ResourceId;
+import org.activityinfo.model.type.enumerated.EnumItem;
+import org.activityinfo.model.type.enumerated.EnumType;
+import org.activityinfo.server.command.DispatcherSync;
+import org.activityinfo.server.command.ResourceLocatorSync;
+import org.activityinfo.server.endpoint.rest.RootResource;
+import org.activityinfo.service.feed.FeedService;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.logging.Logger;
+
+import static java.util.logging.Level.SEVERE;
+import static javax.ws.rs.core.MediaType.APPLICATION_FORM_URLENCODED;
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+import static javax.ws.rs.core.Response.Status.UNAUTHORIZED;
+
+@Path("/AkvoFlow")
+public class AkvoFlowResource {
+    private static final Logger LOGGER = Logger.getLogger(AkvoFlowResource.class.getName());
+
+    private final FeedService feedService;
+    private final DispatcherSync dispatcher;
+    private final ResourceLocatorSync locator;
+
+    @Inject
+    public AkvoFlowResource(ResourceLocatorSync resourceLocatorSync, DispatcherSync dispatcherSync) {
+        locator = resourceLocatorSync;
+        dispatcher = dispatcherSync;
+        feedService = new AkvoFlowFeed(CuidAdapter.activityFormClass(13365), resourceLocatorSync);
+    }
+
+    @POST
+    @Path("/formClass")
+    @Consumes(TEXT_PLAIN)
+    public Response importFormClass(@Context UriInfo uri, String input) {
```

Comment:
Processing the input should really be done within the context of task-- with a large number of updates or a high remote latency you still risk exceeding the request timeout even with if datastore writes are pushed to the task queue.